packages/runtime/src/builder/serialization/control-serialization.ts

```diff
@@ -1040,12 +1049,17 @@ export function serializeControls(
 }
 
 export function deserializeControls(
-  serializedControls: Record<string, SerializedControl>,
-  { onError }: { onError?: (err: Error) => void } = {},
+  serializedControls: Record<string, unknown>,
+  {
+    onError,
+  }: { onError?: (err: Error, data: { key: string; serializedControl: unknown }) => void } = {},
 ): Record<string, DeserializedControl> {
   return Object.entries(serializedControls).reduce(
     (deserializedControls, [key, serializedControl]) => {
       try {
+        if (!isSerializedControl(serializedControl)) {
+          throw new Error(`Expected serialized control for ${key}, received ${serializedControl}`)
```

Comment:
Since this one is going to be caught and prefixed with `Could not deserialize control for "${key}":` below, I think we can just say `Expected serialized control data, got ${JSON.strinfigy(serializedControl)}`.